### ISSUE TITLE
Music Integration

### ISSUE THREAD

**Issue by RokeJulianLockhart** [FossifyOrg/Clock#43]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Clock/issues) or [closed](https://github.com/FossifyOrg/Clock/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Clock/discussions) or [closed](https://github.com/FossifyOrg/Clock/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Clock/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

I want a way to play a different (random) song of my choice every day. If Clock invoked the `documentsui` File Picker, I could use 

### Why do you want this feature?

I sleep through every alarm if they repeat each day. They must be different each day for me to wake.

### Additional information

As https://newsroom.spotify.com/2018-07-31/wake-up-to-the-perfect-soundtrack-with-spotify-and-clock-app-from-google/ explains, https://www.apkmirror.com/apk/google-inc/clock/clock-7-7-release/google-clock-7-7-602251855-3-android-apk-download/ provides the ability to choose a random music track from YouTube Music or Spotify. I use it every day. I won't be able to use Fossify Clock without an alternative.

**Comment by Roccovdberg** [comment]
Hi! I am Rocco, a student from the Netherlands, and I would like to work on this feature!

**Comment by Aga-C** [comment]
@Roccovdberg 

![image.png](https://github.com/FossifyOrg/Clock/assets/85929121/a234500b-7a63-4e62-bc37-871ba2d50368)

**Comment by Roccovdberg** [comment]
Oh okay thankyou

**Comment by drhirn** [comment]
No use for Spotify or Youtube here, but I would love to wake up listening to a randomly generated playlist from my locally stored music files.
In fact, I'm now searching for years for an app which can do this and is still in development.

Would be willing to contribute financially for this feature.

**Comment by e-t-l** [comment]
Integrating with Spoify or Youtube directly might be tricky (I mean, it's actually not because both apps offer a protocol that makes it easy to command them to play music) but to make this both easier for the devs and more versatile for power users, it would be sufficient to simply allow this app to send a custom Intent when an alarm rings, is snoozed, and is dismissed. It would be the user's responsibility to correctly format the intent to command Spotify (or any music app of their choice, really) to start/stop playing.

**Comment by e-t-l** [comment]
> integrating the correct intent would be trivial for the developer – they're adequately stable.

True, I was thinking in terms of providing maximum versatility. As it's often power users who prefer Fossify and degoogling, folks may not be satisfied with an implementation that copies the current Google Clock app—i.e. choosing to play music from Spotify, Youtube, or a limited list of other apps—and would prefer the ability to configure alarms to trigger ANY music app (or any other app, really) to their hearts' desire.

> I don't know whether this functionality is exposed as an intent, though.

It is!
- [StackOverflow: "Is there a way to start playing music on Spotify with an intent?"](http://stackoverflow.com/questions/53969855/ddg#57600180) 
- [Spotify Android SDK for Developers](https://developer.spotify.com/documentation/android)

**Comment by e-t-l** [comment]
I don't think ChooserActivity or a HTTPS calls (to youtube.com, open.spotify.com, etc) would work well, because this is something most users would likely want their alarm sound to happen without launching a new app, and with the phone locked.

I'm no developer either, but if you look at the SDK I linked, I believe the Spotify app registers an actual `spotify:` URI protocol, which has a straightforward syntax for specifying what track/playlist to play. I imagine the Youtube Music app does something similar.

**Comment by e-t-l** [comment]
I'm confused by what you're asking. It's an intent, there's nothing more generic than that. There's no "special case" involved. Did you read the links I shared?

**Comment by RokeJulianLockhart** [comment]
@e-t-l, both would be ideal! However, I know that some maintainers want to ensure that they don't depart far from the original “simple” philosophy of its predecessor.
